AWS-CodePipeline/sam-monorepo-pipeline-template/{{cookiecutter.outputDir}}/codepipeline.yaml

```diff
@@ -0,0 +1,518 @@
+  TestingCloudFormationExecutionRole:
+    Type: String
+    Default: "{{cookiecutter.test_cloudformation_execution_role}}"
+  # Stage 2 Parameters
+  ProdConfigEnvName:
+    Type: String
+    Default: "{{cookiecutter.prod_stage_name}}"
+  ProdPipelineExecutionRole:
+    Type: String
+    Default: "{{cookiecutter.prod_pipeline_execution_role}}"
+  ProdCloudFormationExecutionExeRole:
+    Type: String
+    Default: "{{cookiecutter.prod_cloudformation_execution_role}}"
+
+Resources:
+  #  ____  _            _ _
+  # |  _ \(_)_ __   ___| (_)_ __   ___
+  # | |_) | | '_ \ / _ | | | '_ \ / _ \
+  # |  __/| | |_) |  __| | | | | |  __/
+  # |_|   |_| .__/ \___|_|_|_| |_|\___|
+  #         |_|
+  Pipeline:
+    Type: AWS::CodePipeline::Pipeline
+    Properties:
+      ArtifactStore:
+        Location: !Ref PipelineArtifactsBucket
+        Type: S3
+      RoleArn: !GetAtt CodePipelineExecutionRole.Arn
+      RestartExecutionOnUpdate: false
+      Stages:
+        - Name: Source
+          Actions:
+            - Name: SourceCodeRepo
+              ActionTypeId:
+                Version: "1"
+                Category: Source
+                Owner: AWS
+                Provider: CodeCommit
+              Configuration:
+                BranchName:
+                  Fn::ImportValue: !Sub "${MonorepoRootStack}-MainGitBranch"
+                RepositoryName:
+                  Fn::ImportValue: !Sub "${MonorepoRootStack}-CodeCommitRepositoryName"
+                PollForSourceChanges: false
+              OutputArtifacts:
+                - Name: SourceCodeAsZip
+              RunOrder: 1
+
+        - Name: UpdatePipeline
+          Actions:
+            - Name: CreateChangeSet
+              ActionTypeId:
+                Category: Deploy
+                Owner: AWS
+                Provider: CloudFormation
+                Version: "1"
+              Configuration:
+                ActionMode: CHANGE_SET_REPLACE
+                RoleArn: !GetAtt PipelineStackCloudFormationExecutionRole.Arn
+                StackName: !Ref AWS::StackName
+                ChangeSetName: !Sub ${AWS::StackName}-ChangeSet
+                TemplatePath: !Sub "SourceCodeAsZip::${ProjectSubfolder}/codepipeline.yaml"
+                Capabilities: CAPABILITY_NAMED_IAM
+              InputArtifacts:
+                - Name: SourceCodeAsZip
+              RunOrder: 1
+            - Name: ExecuteChangeSet
+              ActionTypeId:
+                Category: Deploy
+                Owner: AWS
+                Provider: CloudFormation
+                Version: "1"
+              Configuration:
+                ActionMode: CHANGE_SET_EXECUTE
+                RoleArn: !GetAtt PipelineStackCloudFormationExecutionRole.Arn
+                StackName: !Ref AWS::StackName
+                ChangeSetName: !Sub ${AWS::StackName}-ChangeSet
+              OutputArtifacts:
+                - Name: !Sub ${AWS::StackName}ChangeSet
+              RunOrder: 2
+
+        - Name: UnitTest
```

Comment:
nit: in codepipeline template, we commented it out with
> Uncomment and modify the following step for running the unit-tests

Maybe we can make it consistent with the code pipeline template. If you think this should be on by default, we can uncomment them in all the templates in another PR.

Reply:
IMHO it should be on by default... I believe it doesn't add _too much_ overhead, and suggests a best practice of having tests and running them :)
I can raise a separate PR for the other template with this part uncommented...

Reply:
Yeah it is not blocking then